src/config/backend/EdFi.DmsConfigurationService.Backend.Postgresql/VendorRepository.cs

```diff
@@ -188,5 +188,76 @@ public async Task<DeleteResult> DeleteAsync(long id)
                 return new DeleteResult.UnknownFailure(ex.Message);
             }
         }
+
+        public async Task<GetResult<Vendor>> GetVendorByIdWithApplicationsAsync(long vendorId)
+        {
+            string sql = """
+                    SELECT v.Company, v.ContactName, v.ContactEmailAddress,
+                        a.Id, a.ApplicationName, a.VendorId, a.ClaimSetName, 
+                        eo.EducationOrganizationId
+                    FROM dmscs.vendor v
+                    LEFT OUTER JOIN dmscs.Application a ON v.Id = a.VendorId
+                    LEFT OUTER JOIN dmscs.ApplicationEducationOrganization eo ON a.Id = eo.ApplicationId
+                    WHERE v.Id = @VendorId;
+                """;
+
+            await using var connection = new NpgsqlConnection(databaseOptions.Value.DatabaseConnection);
+
+            try
+            {
+                var vendors = await connection.QueryAsync<Vendor, Application, long?, Vendor>(
+                    sql,
+                    (vendor, application, educationOrganizationId) =>
+                    {
+                        var existingApplication = vendor.Applications
+                            .FirstOrDefault(app => app.Id == application?.Id);
+
+                        if (existingApplication == null && application != null)
```

Comment:
`application` cannot be null here. You can remove the check. Also remove the `?` on line 213